packages/dds/map/src/directory.ts

```diff
@@ -1551,7 +1558,12 @@ class SubDirectory extends TypedEventEmitter<IDirectoryEvents> implements IDirec
         // This should make the subdirectory structure unreachable so it can be GC'd and won't appear in snapshots
         // Might want to consider cleaning out the structure more exhaustively though?
         const successfullyRemoved = this._subdirectories.delete(subdirName);
-        this.disposeSubDirectoryTree(previousValue);
+        if (previousValue !== undefined) {
+            this.disposeSubDirectoryTree(previousValue);
```

Comment:
Yeah, since the children will be disposed too, they can listen to dispose events which tells that no other operation is now permitted on them now. The "subDirectoryDeleted" event will be fired only the ancestors of the deleted directory.

Reply:
This issue seems to have been resolved with the recent changes, am I correct?